bearium/sources/clearinghouse.move

```diff
@@ -115,6 +109,39 @@ module bearium::room {
         profit
     }
 
+    //--------------
+    // Agentic Hooks
+    //--------------
+
+    public fun register_agent<ORIGIN>(code_owner: &signer, f: Dispatch) acquires Agency {
+        let builder = signer::address_of(code_owner);
+        enforce_code_object_owner<ORIGIN>(builder);
+        let og = type_to_address<ORIGIN>();
+        let registry = &mut Agency[@bearium].registry;
+        table::add(registry, og, f);
+    }
+
+    fun enforce_code_object_owner<ORIGIN>(caller: address) {
+        let code = type_to_address<ORIGIN>();
+        
+        // deploy module under an account
+        if (caller == code) return;
+        
+        // deploy module under an object
+        let code_object = object::address_to_object<ObjectCore>(code);
+        assert!(object::is_owner(code_object, caller));
```

Comment:
I suggest you take in the signer reference here, so that you don't accidentally enforce it on a non-authorized address